src/hooks/useReactEventPlugin.tsx

```diff
@@ -0,0 +1,51 @@
+import type { DOMEventMap } from "prosemirror-view";
+import { useCallback, useMemo, useState } from "react";
+
+import { createReactEventPlugin } from "../plugins/createReactEventPlugin.js";
+import type { EventHandler } from "../plugins/createReactEventPlugin.js";
+
+/**
+ * Produces a plugin that can be used with ProseMirror to handle DOM
+ * events at the EditorView.dom element.
+ *
+ * - `reactEventsPlugin` is a ProseMirror plugin for handling DOM events
+ * at the EditorView.dom element. It should be passed to `useEditorView`,
+ * along with any other plugins.
+ *
+ * - `registerEventListener` and `unregisterEventListener` should be
+ * passed to `EditorProvider`.
+ */
+export function useReactEventPlugin() {
+  const [registry, setRegistry] = useState(
+    new Map<keyof DOMEventMap, Set<EventHandler>>()
+  );
+
+  const registerEventListener = useCallback(
+    (eventType: keyof DOMEventMap, handler: EventHandler) => {
+      const handlers = registry.get(eventType) ?? new Set<EventHandler>();
+      handlers.add(handler);
+      if (!registry.has(eventType)) {
+        // We only need to force an update when the set of event types
+        // that we care about changes
```

Comment:
This comment still makes sense, I think, but "force an update" makes less sense without `forceUpdate`. May or may not need the comment at all.